File: Scala/FindMinimumInRotatedArray/src/solution/SolutionTest.scala
package solution

import org.scalatest.FunSuite

/**
 * Created by zhangyi on 2/19/15.
 */

class SolutionTest extends FunSuite {
  val sol = new Solution
  test("Test single element") {
    val num1 = Array(0)
    assert(sol.finMin(num1) == 0)
  }
  test("Test rotated array with two elements") {
    val num2 = Array(2, 1)
    assert(sol.finMin(num2) == 1)
  }
  test("Test rotated array with duplicate elements") {
    val num3 = Array(1, 0, 0, 0, 0)
    assert(sol.finMin(num3) == 0)
  }
  test("Test unrotated array") {
    val num4 = Array(1, 2, 3, 4, 4, 5, 6)
    assert(sol.finMin(num4) == 1)
  }
}

File: Java/FindMinimumInRotatedSortedArray_Dup/src/solution/SolutionTest.java
package solution;

import static org.junit.Assert.*;

public class SolutionTest {

    @org.junit.Test
    public void testFinMin() throws Exception {
        Solution sol = new Solution();
        //sol.findMin(test1);
        // test single element array
        int[] test2 = {1};
        assertEquals(sol.findMin(test2), 1);
        // test unrotated array
        int[] test3 = {1, 2, 3, 4, 5, 6};
        assertEquals(sol.findMin(test3), 1);
        // test rotate one number
        int[] test4 = {2, 3, 4, 5, 6, 1};
        assertEquals(sol.findMin(test4), 1);
        // test rotate last number
        int[] test5 = {6, 1, 2, 3, 4, 5};
        assertEquals(sol.findMin(test5), 1);
        // test duplicate
        int[] test6 = {2, 2, 2, 2, 1, 2, 2};
        assertEquals(sol.findMin(test6), 1);
        // test duplicate
        int[] test7 = {1, 2, 2, 2, 2, 2, 2};
        assertEquals(sol.findMin(test7), 1);
        // test duplicate
        int[] test8 = {2, 0, 1, 1, 1};
        assertEquals(sol.findMin(test8), 0);
        // test duplicate

        int[] test9 = {1, 1};
        assertEquals(sol.findMin(test9), 1);
    }
}